nexus/src/app/mod.rs

```diff
@@ -0,0 +1,471 @@
+    ///
+    /// async fn my_thing_fetch(
+    ///     nexus: &Nexus,
+    ///     datastore: &DataStore,
+    ///     opctx: &OpContext,
+    ///     organization_name: &Name,
+    ///     the_name: &Name,
+    /// ) -> Result<(), Error>
+    /// {
+    ///     // You will want to have defined your own ResourceType variant for
+    ///     // this resource, even though it's still a stub.
+    ///     let resource_type: ResourceType = todo!();
+    ///     let lookup_type = LookupType::ByName(the_name.to_string());
+    ///     let not_found_error = lookup_type.into_not_found(resource_type);
+    ///     let unimp = Unimpl::ProtectedLookup(not_found_error);
+    ///     Err(nexus.unimplemented_todo(opctx, unimp).await)
+    /// }
+    /// ```
+    pub async fn unimplemented_todo(
+        &self,
+        opctx: &OpContext,
+        visibility: Unimpl,
+    ) -> Error {
+        // Deny access to non-super-users.  This is really just for the benefit
+        // of the authz coverage tests.  By requiring (and testing) correct
+        // authz behavior for stubs, we ensure that that behavior is preserved
+        // when the stub's implementation is fleshed out.
+        match opctx.authorize(authz::Action::Modify, &authz::FLEET).await {
+            Err(error @ Error::Forbidden) => {
+                // Emulate the behavior of `Authz::authorize()`: if this is a
+                // non-public resource, then the user should get a 404, not a
+                // 403, when authorization fails.
+                if let Unimpl::ProtectedLookup(lookup_error) = visibility {
+                    lookup_error
+                } else {
+                    error
+                }
+            }
+            Err(error) => error,
+            Ok(_) => {
+                // In the event that a superuser actually gets this far, produce
+                // a server error.
+                //
+                // It's tempting to use other status codes here:
+                //
+                // "501 Not Implemented" is specifically when we don't recognize
+                // the HTTP method and cannot implement it on _any_ resource.
+                //
+                // "405 Method Not Allowed" is specifically when an HTTP method
+                // isn't supported.  That doesn't feel quite right either --
+                // this is usually interpreted to mean "not part of the API",
+                // which it obviously _is_, since the client found it in the API
+                // spec.
+                //
+                // Neither of these is true: this HTTP method on this HTTP
+                // resource is part of the API, and it will be supported by the
+                // server, but it doesn't work yet.
+                Error::internal_error("endpoint is not implemented")
+            }
+        }
+    }
+
+    pub fn datastore(&self) -> &Arc<db::DataStore> {
+        &self.db_datastore
+    }
+
+    // Built-in users
```

Comment:
Built-in users + roles *could* go somewhere else, but given that they're built-in, I'm not totally sure where to place them. Could be here? Could be in `silo.rs`? Could be somewhere else?

Reply:
iam.rs?

Reply:
SGTM - I'll rename `silo.rs` to `iam.rs` and include it there, which I think also addresses your comment below.

Reply:
Cool.  I wouldn't actually put silos with IAM though.  They're going to be complicated enough on their own when we add IdPs.

Reply:
Okay, splitting up by "silo + user + ssh keys (which seem coupled to users)" in silo.rs and "built ins" in iam.rs